arelle/plugin/validate/DBA/rules/tm.py

```diff
@@ -22,6 +22,50 @@
 _: TypeGetText
 
 
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+)
+def rule_tm13(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation]:
+    """
+    DBA.TM13: gsd:IdentificationNumberCvrOfReportingEntity must only be tagged once if tagged
+    """
+    modelXbrl = val.modelXbrl
+    facts = modelXbrl.factsByQname.get(pluginData.identificationNumberCvrOfReportingEntityQn)
+    if facts is not None and len(facts) > 1:
+        yield Validation.error(
+            'DBA.TM13',
+            _('IdentificationNumberCvrOfReportingEntity must only be tagged once. {} facts were found.').format(len(facts)),
+            modelObject=facts
+        )
+
+
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+)
+def rule_tm16(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation]:
+    """
+    DBA.TM16: gsd:InformationOnTypeOfSubmittedReport must only be tagged once if tagged
+    """
+    modelXbrl = val.modelXbrl
+    facts = modelXbrl.factsByQname.get(pluginData.informationOnTypeOfSubmittedReportQn)
+    if facts is not None and len(facts) > 1:
```

Comment:
Arbitrary line.  If all of the checks are essentially running the same check with different qnames, wouldn't it be easier to just write a function in the init similar to errorOnRequiredFact to handle these?

Reply:
Its a fair argument. The check itself was so elementary that I didnt feel like it was worth condensing it down to a helper function.